sway/commands/resize.c

```diff
@@ -292,6 +292,10 @@ static struct cmd_results *resize_adjust_tiled(uint32_t axis,
  */
 static struct cmd_results *resize_set_tiled(struct sway_container *con,
 		struct movement_amount *width, struct movement_amount *height) {
+	if (container_is_scratchpad_hidden_or_child(con)) {
```

Comment:
Refusing in `resize_set_tiled` won't handle the grow/shrink resize incantations. E.g. try growing a scratchpad hidden split container perpendicular to the parent layout and you can still repro the crash.

<details><summary>bt</summary>

```
#0  0x00005555555d8c4d in list_find (list=0x0, item=0x555555a240c0) at ../common/list.c:71
#1  0x00005555555c7fc8 in container_sibling_index (child=0x555555a240c0) at ../sway/tree/container.c:1386
#2  0x00005555555b2e1a in container_find_resize_parent (con=0x555555a240c0, axis=12) at ../sway/commands/resize.c:54
#3  0x00005555555b2eac in container_resize_tiled (con=0x555555a1dfe0, axis=12, amount=10) at ../sway/commands/resize.c:72
#4  0x00005555555b39d7 in resize_adjust_tiled (axis=12, amount=0x7fffffffd690) at ../sway/commands/resize.c:282
#5  0x00005555555b4951 in cmd_resize_adjust (argc=0, argv=0x555555a92730, multiplier=1) at ../sway/commands/resize.c:556
#6  0x00005555555b4a92 in cmd_resize (argc=3, argv=0x555555a92718) at ../sway/commands/resize.c:579
#7  0x000055555556c481 in execute_command (_exec=0x5555557384e0 "[con_mark=test] resize grow width 10px", seat=0x555555713710, con=0x0) at ../sway/commands.c:309
#8  0x0000555555578f41 in ipc_client_handle_command (client=0x555555a22180, payload_length=38, payload_type=IPC_COMMAND) at ../sway/ipc-server.c:651
#9  0x0000555555577a9f in ipc_client_handle_readable (client_fd=56, mask=1, data=0x555555a22180) at ../sway/ipc-server.c:262
#10 0x00007ffff7805ae2 in wl_event_loop_dispatch (loop=0x555555612050, timeout=timeout@entry=-1) at ../wayland-1.22.0/src/event-loop.c:1027
#11 0x00007ffff78062d7 in wl_display_run (display=0x555555611f60) at ../wayland-1.22.0/src/wayland-server.c:1493
#12 0x000055555557cbac in server_run (server=0x555555601b40 <server>) at ../sway/server.c:337
#13 0x000055555557bb67 in main (argc=3, argv=0x7fffffffde78) at ../sway/main.c:415
```
</details>

Refusing in `container_resize_tiled` as suggested effectively removes all broken (and working) cases of resize with scratchpad hidden split containers, but I'm not sure it's a feature we care to keep. Resize of normal scratchpad windows will still be handled as part of the floating case, as there are otherwise no tiled containers in the scratchpad.

Reply:
Hi @rpigott, thanks for the feedback. I updated the patch.

I'm unsure on what `cmd_status` to use (`CMD_FAILURE` vs `CMD_INVALID`)

Reply:
When the request is not serviceable or it doesn't make sense with the layout, please use CMD_FAILURE. CMD_INVALID is for when we don't understand what is requested.